ballerina/modules/serializer/node.bal

```diff
@@ -14,18 +14,17 @@
 
 import yaml.common;
 import yaml.schema;
-
-const string INVALID_PLANAR_PATTERN = "([\\w|\\s]*[\\-|\\?|:|] [\\w|\\s]*)|"
-    + "([\\w|\\s]* #[\\w|\\s]*)|"
-    + "([,|\\[|\\]|\\{|\\}|&\\*|!\\||>|'|\"|%|@|`][\\w|\\s]*)";
+import yaml.parser;
+import ballerina/lang.regexp;
 
 isolated function serializeString(SerializerState state, json data, string tag) {
     string value = data.toString();
-    state.events.push({
-        value: re `${INVALID_PLANAR_PATTERN}`.isFullMatch(value) || state.forceQuotes
-                ? string `${state.delimiter}${value}${state.delimiter}` : value,
-        tag
-    });
+    if value.includes("\n") {
```

Comment:
```suggestion
    if value.includes("\n" || value.includes("\r\n") {
```

Reply:
Supporting OS-specific new lines is tracked with another issue: https://github.com/ballerina-platform/ballerina-library/issues/4330.